Assets/LeapMotion/Core/Scripts/Streams/PoseStreams/PoseStreamMovingAverageFilter.cs

```diff
@@ -0,0 +1,92 @@
+﻿using System;
+using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+
+namespace Leap.Unity.Streams {
+
+  public class PoseStreamMovingAverageFilter : MonoBehaviour,
+                                               IStreamReceiver<Pose>,
+                                               IStream<Pose> {
+
+    [Range(0, 16)]
+    public int neighborRadius = 4;
+
+    public event Action OnOpen  = () => { };
+    public event Action<Pose> OnSend = (pose) => { };
+    public event Action OnClose = () => { };
+
+    private RingBuffer<Pose> buffer;
+
+    private int totalWidth { get { return neighborRadius * 2 + 1; } }
+
+    public void Open() {
+      if (buffer == null || buffer.Capacity != totalWidth) {
+        buffer = new RingBuffer<Pose>(totalWidth);
+      }
+      buffer.Clear();
+
+      OnOpen();
+    }
+
+    public void Receive(Pose data) {
+      if (buffer.Capacity == 0) {
+        OnSend(data);
+      }
+
+      bool bufferWasNotFull = false;
+      if (!buffer.IsFull) {
+        bufferWasNotFull = true;
+      }
+
+      buffer.Add(data);
+
+      if (buffer.IsFull) {
+        if (bufferWasNotFull) {
+          for (int i = 1; i < buffer.Count; i += 2) {
```

Comment:
Could you explain the logic here?  It seems that the stream doesn't send anything until the buffer fills up, but once it does, it quickly outputs one value for every value in the buffer?  Why not be sending averages as the buffer fills up?